src/platform/nm-platform-utils.c

```diff
@@ -395,6 +429,8 @@ nmp_utils_ethtool_set_link_settings (int ifindex,
 	};
 
 	g_return_val_if_fail (ifindex > 0, FALSE);
+	g_return_val_if_fail (   (!speed && duplex != NM_PLATFORM_LINK_DUPLEX_UNKNOWN)
+	                      || (speed && duplex == NM_PLATFORM_LINK_DUPLEX_UNKNOWN), FALSE);
```

Comment:
The condition still seems inverted to me: it should allow only both values set or both unset.

Reply:
Right, fixed thanks